src/ledger/LedgerTxn.cpp

```diff
@@ -1089,9 +1184,19 @@ LedgerTxn::Impl::loadBestOffer(LedgerTxn& self, Asset const& buying,
     throwIfSealed();
     throwIfChild();
 
-    std::unordered_set<LedgerKey> exclude;
-    auto le = getBestOffer(buying, selling, exclude);
-    return le ? load(self, LedgerEntryKey(*le)) : LedgerTxnEntry();
+    auto le = getBestOffer(buying, selling);
+    auto res = le ? load(self, LedgerEntryKey(*le)) : LedgerTxnEntry();
+
+    try
+    {
+        AssetPair assets{buying, selling};
+        mWorstBestOffer[assets] = le;
```

Comment:
I don't know if we can give `mWorstBestOffer` this value: `le` may have as value some random entry from `mEntry` (via `mOrderBook`).

If this happens, this will cause `loadBestOffer` to return duplicates, where as the previous implementation would never do this as whatever was in `mEntry` was always excluded.

I don't know if we should detect this. I guess the reason is that `getBestOffer` has a different behavior now in this situation. I don't think the current code cares (we call `getBestOffer` right after creating a `LedgerTxn` scope)